### PR TITLE
Add Support to User Assigned Identities

### LINKED ISSUE #3: [FEAT] Add Support to User Managed Identity
### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The Module only supports System assigned Managed identity. I would like have the possibility also add User Managed Identities.

Also, I think the container_registry_managed_identity_client_id parameter don't have any effect/meaning if the managed identity passed in this parameter won't be created as User Managed Identity on Web App.

With this, i identity two new capabilities related with User Managed Identities:

1. Add possibility to create user managed identity. This will required new variables
2. Create automatically an user managed identity when the container_registry_managed_identity_client_id is not null. Don't need new variables.

### New or Affected Resource(s)/Data Source(s)

azurerm_linux_web_app (container and linux module) and azurerm_windows_web_app

### Potential Terraform Configuration

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by fdmsantos**
Fixes #3.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Changes proposed in this pull request
- Add possibility to add User Assigned Identities

@claranet/fr-azure-reviewers

### CODE REVIEW

**Comment by Shr3ps** [comment]
Thanks for this PR :)
Can you also update READMEs in each sub-module folder?

**Review by Shr3ps — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+# Unreleased
+
+Added
+    * [GH-4](https://github.com/claranet/terraform-azurerm-app-service/pull/4): Add Support to User Assigned Identities
```
Comment (zfiel) [inline CHANGELOG.md]:
```suggestion
Added
    * [GH-4](https://github.com/claranet/terraform-azurerm-app-service/pull/4): Add support for User Assigned Identities
```

File: README.md [hunk 2]
```diff
@@ -228,6 +228,7 @@ No resources.
 | environment | Project environment | `string` | n/a | yes |
 | extra\_tags | Extra tags to add. | `map(string)` | `{}` | no |
 | https\_only | HTTPS restriction for App Service. See documentation https://www.terraform.io/docs/providers/azurerm/r/app_service.html#https_only | `bool` | `false` | no |
+| identity | Map with identity block informations. | <pre>object({<br>    type         = string<br>    identity_ids = list(string)<br>  })</pre> | <pre>{<br>  "identity_ids": [],<br>  "type": "SystemAssigned"<br>}</pre> | no |
```
Comment (zfiel) [inline README.md]:
```suggestion
| identity | Map with identity block information. | <pre>object({<br>    type         = string<br>    identity_ids = list(string)<br>  })</pre> | <pre>{<br>  "identity_ids": [],<br>  "type": "SystemAssigned"<br>}</pre> | no |
```

File: modules/container-web-app/README.md [hunk 3]
```diff
@@ -264,6 +264,7 @@ module "container_web_app" {
 | environment | Project environment | `string` | n/a | yes |
 | extra\_tags | Extra tags to add. | `map(string)` | `{}` | no |
 | https\_only | HTTPS restriction for App Service. See documentation https://www.terraform.io/docs/providers/azurerm/r/app_service.html#https_only | `bool` | `false` | no |
+| identity | Map with identity block informations. | <pre>object({<br>    type         = string<br>    identity_ids = list(string)<br>  })</pre> | <pre>{<br>  "identity_ids": [],<br>  "type": "SystemAssigned"<br>}</pre> | no |
```
Comment (zfiel) [inline modules/container-web-app/README.md]:
```suggestion
| identity | Map with identity block information. | <pre>object({<br>    type         = string<br>    identity_ids = list(string)<br>  })</pre> | <pre>{<br>  "identity_ids": [],<br>  "type": "SystemAssigned"<br>}</pre> | no |
```

File: modules/container-web-app/variables.tf [hunk 4]
```diff
@@ -291,3 +291,15 @@ variable "app_service_logs" {
   })
   default = null
 }
+
+variable "identity" {
+  description = "Map with identity block informations."
```
Comment (zfiel) [inline modules/container-web-app/variables.tf]:
```suggestion
  description = "Map with identity block information."
```

File: modules/linux-web-app/README.md [hunk 5]
```diff
@@ -259,6 +259,7 @@ module "linux_web_app" {
 | environment | Project environment | `string` | n/a | yes |
 | extra\_tags | Extra tags to add. | `map(string)` | `{}` | no |
 | https\_only | HTTPS restriction for App Service. See documentation https://www.terraform.io/docs/providers/azurerm/r/app_service.html#https_only | `bool` | `false` | no |
+| identity | Map with identity block informations. | <pre>object({<br>    type         = string<br>    identity_ids = list(string)<br>  })</pre> | <pre>{<br>  "identity_ids": [],<br>  "type": "SystemAssigned"<br>}</pre> | no |
```
Comment (zfiel) [inline modules/linux-web-app/README.md]:
```suggestion
| identity | Map with identity block information. | <pre>object({<br>    type         = string<br>    identity_ids = list(string)<br>  })</pre> | <pre>{<br>  "identity_ids": [],<br>  "type": "SystemAssigned"<br>}</pre> | no |
```

File: modules/linux-web-app/variables.tf [hunk 6]
```diff
@@ -282,3 +282,15 @@ variable "app_service_logs" {
   })
   default = null
 }
+
+variable "identity" {
+  description = "Map with identity block informations."
```
Comment (zfiel) [inline modules/linux-web-app/variables.tf]:
```suggestion
  description = "Map with identity block information."
```

File: modules/windows-web-app/README.md [hunk 7]
```diff
@@ -260,6 +260,7 @@ module "windows_web_app" {
 | environment | Project environment | `string` | n/a | yes |
 | extra\_tags | Extra tags to add. | `map(string)` | `{}` | no |
 | https\_only | HTTPS restriction for App Service. See documentation https://www.terraform.io/docs/providers/azurerm/r/app_service.html#https_only | `bool` | `false` | no |
+| identity | Map with identity block informations. | <pre>object({<br>    type         = string<br>    identity_ids = list(string)<br>  })</pre> | <pre>{<br>  "identity_ids": [],<br>  "type": "SystemAssigned"<br>}</pre> | no |
```
Comment (zfiel) [inline modules/windows-web-app/README.md]:
```suggestion
| identity | Map with identity block information. | <pre>object({<br>    type         = string<br>    identity_ids = list(string)<br>  })</pre> | <pre>{<br>  "identity_ids": [],<br>  "type": "SystemAssigned"<br>}</pre> | no |
```

File: modules/windows-web-app/variables.tf [hunk 8]
```diff
@@ -282,3 +282,15 @@ variable "app_service_logs" {
   })
   default = null
 }
+
+variable "identity" {
+  description = "Map with identity block informations."
```
Comment (zfiel) [inline modules/windows-web-app/variables.tf]:
```suggestion
  description = "Map with identity block information."
```

File: variables.tf [hunk 9]
```diff
@@ -286,3 +286,15 @@ variable "app_service_logs" {
   })
   default = null
 }
+
+variable "identity" {
+  description = "Map with identity block informations."
```
Comment (zfiel) [inline variables.tf]:
```suggestion
  description = "Map with identity block information."
```

**Review by zfiel — APPROVED** [review]

**Review by rossifumax — APPROVED** [review]

**Review by maxpoullain — APPROVED** [review]

**Pull request merged**